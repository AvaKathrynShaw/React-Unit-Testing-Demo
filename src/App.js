import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';


/*
class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;

*/


import React, { Component } from 'react'
import './App.css'
import { getUser } from './api/github'

const renderLine = (user, key) => <li key={key}><b>{key}</b>: {user[key]}</li>

class App extends Component {
  constructor (props) {
    super(props)
    this.state = { user: {} }
  }

  componentDidMount () {
    getUser('vnglst').then(data => {
      this.setState({ user: data.entity })
    })
  }

  render () {
    const { user } = this.state
    return (
      <div className='App'>
        <ul style={{ listStyle: 'none' }}>
          {
            // Loop over the object keys and render each key
            Object.keys(user).map(key => renderLine(user, key))
          }
        </ul>
      </div>
    )
  }
}

export default App
