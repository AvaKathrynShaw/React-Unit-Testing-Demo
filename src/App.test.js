import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Component: App', () => {
  const items = ['Learn react', 'rest', 'go out'];

  it('should match its empty snapshot', () => {
    const tree = renderer.create(
      <App />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should add an item based on the value in the state', () => {
  	const component = shallow(<App />);
    const preventDefault = jest.fn();
    component.setState({
      items
    });
    component.find('form').simulate('submit', { preventDefault });
  	expect(toJson(component)).toMatchSnapshot();
    expect(preventDefault).toBeCalled();
  });

  it('should pass a selected value to the onChange function', () => {
    const tree = renderer.create(
      <App />
    );
    const component = shallow(<App items={items} />);
    component.find('input').simulate('change', { target: {
      value: 'Change function' }
    });

    expect(toJson(component)).toMatchSnapshot();
  });
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});
