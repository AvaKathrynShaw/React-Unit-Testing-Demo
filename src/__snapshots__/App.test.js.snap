// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should add an item based on the value in the state 1`] = `
<div>
  <form
    className="App"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button>
      Submit
    </button>
  </form>
  <List
    items={
      Array [
        "Learn react",
        "rest",
        "go out",
        "",
      ]
    }
  />
</div>
`;

exports[`Component: App should match its empty snapshot 1`] = `
<div>
  <form
    className="App"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button>
      Submit
    </button>
  </form>
  <ul />
</div>
`;

exports[`Component: App should pass a selected value to the onChange function 1`] = `
<div>
  <form
    className="App"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value="Change function"
    />
    <button>
      Submit
    </button>
  </form>
  <List
    items={Array []}
  />
</div>
`;
